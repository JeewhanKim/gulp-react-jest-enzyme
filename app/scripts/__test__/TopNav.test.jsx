import React from 'react';
import ReactDOM from 'react-dom';

import TopNav from '../components/TopNav.jsx'

import { configure } from 'enzyme';
import { shallow, mount, render } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

import $ from '../jquery-2.2.0.min.js';
global.$ = global.jQuery = $;


describe('TopNav', () => {
  it('should render correctly', () => {
    const output = shallow(
      <TopNav selectedSection="0" clickChapter="0"/>
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});

