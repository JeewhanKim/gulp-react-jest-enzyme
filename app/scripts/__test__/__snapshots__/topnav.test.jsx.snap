// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopNav should render correctly 1`] = `
<section
  id="nav-main"
>
  <div
    className="tabs"
  >
    <ul>
      <li
        className=""
        onClick={[Function]}
      >
        <a>
          <span
            className="icon is-small"
          >
            <i
              className="fa fa-image"
            />
          </span>
          Section 1
        </a>
      </li>
      <li
        className=""
        onClick={[Function]}
      >
        <a>
          <span
            className="icon is-small"
          >
            <i
              className="fa fa-film"
            />
          </span>
          Section 2
        </a>
      </li>
      <li
        className=""
        onClick={[Function]}
      >
        <a>
          <span
            className="icon is-small"
          >
            <i
              className="fa fa-book"
            />
          </span>
          Section 3
        </a>
      </li>
    </ul>
  </div>
</section>
`;
